sklearn/utils/_testing.py

```diff
@@ -769,3 +772,91 @@ def _convert_container(container, constructor_name, columns_name=None):
         return pd.Index(container)
     elif constructor_name == 'slice':
         return slice(container[0], container[1])
+
+
+def raises(expected_exp_type, match=None, may_pass=False, err_msg=None):
+    """Context manager to ensure exceptions are raised within a code block.
+
+    This is similar to and inspired from pytest.raises, but supports a few
+    other cases.
+
+    This is only intended to be used in estimator_checks.py where we don't
+    want to use pytest. In the rest of the code base, just use pytest.raises
+    instead.
+
+    Parameters
+    ----------
+    excepted_exp_type : Exception or list of Exception
+        The exception that should be raised by the block. If a list, the block
+        should raise one of the exceptions.
+    match : str or list of str, default=None
+        A regex that the exception message should match. If a list, one of
+        the entries must match. If None, match isn't enforced.
+    may_pass : bool, default=False
+        If True, the block is allowed to not raise an exception. Useful in
+        cases where some estimators may support a feature but others must
+        fail with an appropriate error message.
```

Comment:
Could this be "always_pass" which makes this context manager a noop?

Reply:
No, I hope the docstring makes it clear, but let me know otherwise.

Reply:
Just read the implementation. I think the confusing part is that the block can raise an exception and the context manager can also raise an exception. Maybe:

> If True, the context manager will not raise an exception if the code block does not.